app/src/main/java/org/simple/clinic/enterotp/EnterOtpUpdate.kt

```diff
@@ -53,11 +53,12 @@ class EnterOtpUpdate(
       event: LoginUserCompleted
   ): Next<EnterOtpModel, EnterOtpEffect> {
     val updatedModel = model.loginFinished()
+    val loginFailedModel = model.loginFailed()
     return when (val result = event.result) {
       LoginResult.Success -> next(updatedModel, ClearLoginEntry, TriggerSync)
-      is LoginResult.ServerError -> dispatch(FailedLoginOtpAttempt(result), ClearPin)
-      LoginResult.NetworkError -> dispatch(ShowNetworkError, ClearPin)
-      LoginResult.UnexpectedError -> dispatch(ShowUnexpectedError, ClearPin)
+      is LoginResult.ServerError -> next(loginFailedModel, FailedLoginOtpAttempt(result), ClearPin)
+      LoginResult.NetworkError -> next(loginFailedModel, ShowNetworkError, ClearPin)
+      LoginResult.UnexpectedError -> next(loginFailedModel, ShowUnexpectedError, ClearPin)
```

Comment:
This works, but I think this could've been handled a little bit cleaner. We are creating 2 different models here at the same time.